### PR TITLE
Task/COOKS-323: refactor get selected geography name

### PR DESCRIPTION
**Pull request by ryamase**
## Overview: ##
Reduce the technical debt related to generating chart labels in dataUtils.

## Related Jira tickets: ##

* [COOKS-123](https://jira.tacc.utexas.edu/browse/COOKS-123)

## Summary of Changes: ##
- Converted PHR_MSA_County_Data from an array to an object.
- Defined a new function `getCountyName` to be used in `getSelectedGeographyName`
- Refactored `getSelectedGeographyName`
- Added new tests to dataUtils.test.js
- Misc changes: fix vite.config.ts, add /coverage to .gitignore, fix the maltreatment if block in MainChart

## Testing Steps: ##
1. Go to https://cep.test/protx/dash/demographics.
2. Make sure Counties is selected in the Area pulldown menu.
3. Click on the map to select a county and ensure the correct county name appears each time.

1. Go to https://cep.test/protx/dash/maltreatment.
2. Click on the map to select a county and ensure the correct county name appears each time.

1. Try out the new tests: from protx-client directory run `npm run test 'dataUtils.test.js'`

### CODE REVIEW

File: protx-client/src/components/ProTx/components/shared/dataUtils.js [hunk 1]
```diff
@@ -27,60 +27,45 @@ const compareSimplifiedValueType = (observedFeature, valueType) => {
...
+  });
+  const countyName = county ? county['County Name'] : 'Not Found';
+  return countyName;
+};
+
+/**
+ * Get a pertinent geography name for a given Geoid and geography.
  * @param {String} currentGeoid
- * @returns {geographyDisplayName: string}
+ * @param {String} geography
```
Comment (nathanfranklin) [inline protx-client/src/components/ProTx/components/shared/dataUtils.js:47]: swap the order this line with the line above it to match the function signature

File: protx-client/src/components/ProTx/components/charts/MainChart.jsx [hunk 2]
```diff
@@ -129,42 +129,42 @@ function MainChart({
...
+  
+      const dispatch = useDispatch();
+  
+      useEffect(() => {
+        if (selectedGeographicFeature && maltreatmentTypes.length !== 0) {
+          dispatch({
+            type: 'FETCH_PROTX_MALTREATMENT_DISTRIBUTION',
+            payload: {
+              area: geography,
+              selectedArea: selectedGeographicFeatureNameComplete,
```
Comment (nathanfranklin) [inline protx-client/src/components/ProTx/components/charts/MainChart.jsx:153]: I see where we can refactor the backend to no longer need this (`selectedArea`) and just use `geoid` instead.  Adding details in a new jira issue https://jira.tacc.utexas.edu/browse/COOKS-327

**Review by nathanfranklin — CHANGES_REQUESTED** [review]
LGTM. 👍 

Just one thing to consider: https://github.com/TACC/protx-dashboard/pull/43/files#r995278499

File: protx-client/src/components/ProTx/components/shared/dataUtils.test.js [hunk 3]
```diff
@@ -21,3 +25,29 @@ describe('util functions: compareSimplifiedValueType ', () => {
...
+  });
+  it("return the correct region name given the corresponding Geoid and a geography type of 'dfps_region'.", () => {
+    expect(getSelectedGeographyName('dfps_region', '7-Austin')).toEqual(
+      'Austin'
+    );
+  });
+  it("return an emptry string when provided an unexpected geography type (e.g. 'state').", () => {
+    expect(getSelectedGeographyName('state', '48')).toEqual('');
+  });
+});
```
Comment (nathanfranklin) [inline protx-client/src/components/ProTx/components/shared/dataUtils.test.js:53]: 💯  on the tests

**Review by nathanfranklin — APPROVED** [review]
LGTM 👍

**Review by sophia-massie — APPROVED** [review]
LGTM great work!

**Pull request merged**